packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,445 @@
+    const startIndex = PillsUtils.handleRollUp(filterPillsRef);
+    if (isCollapsed) {
+      setUpdatedCount(startIndex);
+      setRollUpCount(React.Children.count(children) - startIndex);
+    } else if (startIndex === React.Children.count(children)) {
+      setRollUpCount(0);
+    }
+  }, [children, isCollapsed]);
+
+  const setTabIndex = (val) => {
+    const currentNode = currentPill.current ? filterPillsRef.current.querySelector(`[id=${currentPill.current}]`) : null;
+    if (currentNode) {
+      currentNode.setAttribute('tabIndex', val);
+    }
+  };
+
+  const focusCurrentNode = () => {
+    const currentNode = currentPill.current ? filterPillsRef.current.querySelector(`[id=${currentPill.current}]`) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
+
+  const handleTabIndex = useCallback(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+
+    // if there is a roll Up pill, set tabindex to -1
+    if (rollUpPill) {
+      PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+    }
+
+    if (pills.length > 0 && focusNode.current < pills.length) {
+      PillsUtils.setPillsTabIndex(pills, '-1');
+      currentPill.current = pills[focusNode.current].id;
+      setTabIndex('0');
+    } else if (isCollapsible && rollUpPill) { // if the first pill is rollUp pill, set rollUp pill tabindex 0
+      currentPill.current = rollUpPill.getAttribute('id');
+      setTabIndex('0');
+    }
+  }, [isCollapsible]);
+
+  const handleResize = useCallback((entries) => {
+    if (!Array.isArray(entries)) {
+      return;
+    }
+
+    setUpdatedCount(React.Children.count(children));
+    setRollUpCount(React.Children.count(children));
+
+    if (isCollapsible) {
+      generateRollUp();
+    }
+
+    handleTabIndex();
+  }, [children, isCollapsible, handleTabIndex, generateRollUp]);
+
+  useLayoutEffect(() => {
+    let observer = new ResizeObserver((entries) => {
+      handleResize(entries);
+    });
+    observer.observe(filterPillsRef.current.parentNode);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, [children, handleResize, isCollapsed]);
+
+  const focusPillsContainer = () => {
+    setContainerTabindex('0');
+    filterPillsRef.current.focus();
+  };
+
+  // When a pill is deleted, focuses the new pill.
+  useEffect(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const startIndex = PillsUtils.handleRollUp(filterPillsRef);
+
+    if (isPillDeleted.current) {
+      if (React.Children.count(children) <= 0) {
+        focusPillsContainer();
+        return;
+      }
+      // if there is a roll Up pill, set tabindex to 0
+      if (rollUpPill) {
+        PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
```

Comment:
Should the tabIndex be 0 based on the comment?

Reply:
Its typo in the comment. Its supposed to be '-1'.

Changed in this commit e79be980bb3defef6e9370f26558523eead90bfa